data_model/src/events/data/events.rs

```diff
@@ -0,0 +1,301 @@
+//! This module contains data events
+
+use super::*;
+
+/// Trait for retrieving id from events
+pub trait IdTrait: Identifiable {
+    fn id(&self) -> &Self::Id;
+}
+
+mod asset {
+    //! This module contains `AssetEvent`, `AssetDefinitionEvent` and its impls
+
+    use super::*;
+
+    #[derive(Clone, PartialEq, Eq, Debug, Decode, Encode, Deserialize, Serialize, IntoSchema)]
+    #[non_exhaustive]
+    #[allow(missing_docs)]
+    pub enum AssetEvent {
+        Created(AssetId),
+        Deleted(AssetId),
+        Increased(AssetId),
+        Decreased(AssetId),
+        MetadataInserted(AssetId),
+        MetadataRemoved(AssetId),
```

Comment:
I'm not sure this event should be called MetadataInserted. Maybe rather StoreInserted?

Reply:
actually, I think there should be only 4 events something like:
```
    pub enum AssetEvent {
        Created(AssetId),
        Deleted(AssetId),
        Added(AssetId),
        Removed(AssetId),
}
```

Reply:
Then what event should produce `SetKeyValue<Asset, Name, Value>` and `RemoveKeyValue<Asset, Name>`?

Reply:
As I see it, asset can be created, deleted, added a quantity or removed a quantity where quantity can be a number or key-value. SetKeyValue is adding a key value to asset which holds key-value entities

Reply:
@Arjentix what happened with this? you just resolved the conversation without a comment

Reply:
Sorry, I misunderstood your comment, I've read it as `Ah I see it`

So you think, it should be `Added` and `Removed` for `SetKeyValue` and `RemoveKeyValue`? In this instructions code there are a lot of `metadata` terminology, is it okay then?